__device-tests__/gutenberg-editor-writing-flow.test.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Internal dependencies
+ */
+const { blockNames } = editorPage;
+
+describe( 'Gutenberg Editor - Writing Flow', () => {
+	describe( 'TC007 - Test format detection under the cursor', () => {
+		it( 'checks that the proper format buttons are selected when the cursor is under', async () => {
+			const paragraphHTML = `<!-- wp:paragraph -->
+            <p><span accessibility-id="bold"><strong>bold</strong></span>
+            <span accessibility-id="bold-italic"><strong><i>bold-italic</i></strong>.</span>
+            <span accessibility-id-"strikethrough"><s>strikethrough</s></span></p>
+            <!-- /wp:paragraph -->`;
+
+			// On a rich-text based component, add bold, italic, strikethrough and link formatted text
+			await editorPage.addNewBlock( blockNames.paragraph );
+			const paragraphBlockElement = await editorPage.getTextBlockAtPosition(
+				blockNames.paragraph
+			);
+
+			await editorPage.typeTextToTextBlock(
+				paragraphBlockElement,
+				paragraphHTML
+			);
+
+			// Select bold text
+			// TODO: verify if this is the best method to select text formatting
+			await editorPage.elementsByAccessibilityId( 'bold' );
+
+			// Check that the proper format buttons get selected
+			// TODO: Identify best method to identify format buttons
+
+			// TODO: Invoke assertion or snapshot here?
```

Comment:
We could either assert the presence of specific styles on the buttons, i.e. find the button with a query and verify the styles are applied, or rely upon a visual snapshot. My opinion is that this is a perfect candidate for a visual snapshot. 

We would need to type formatted text, move the cursor to a specific position within the formatted text, and assert with a visual snapshot. 

WDYT?

Reply:
> We would need to type formatted text, move the cursor to a specific position within the formatted text, and assert with a visual snapshot.

I follow and agree with using a visual snapshot here. 👍 

Could you describe more about the need to move the cursor to a specific position within the text, and what the expected result might be? I recognize that we could move the cursor via the various [`element.getLocation`](https://github.com/WordPress/gutenberg/blob/dfdfaa3135b32af9018a41b832c2538c49adf747/packages/react-native-editor/__device-tests__/helpers/utils.js#L336-L344)-based helpers, but don't quite follow the assertion we're looking for in doing so.

Reply:
Thanks for clarifying. I've updated the test to use `selectTextFromElement`, copying the technique from [gutenberg-editor-sanity-test-4](https://github.com/wordpress-mobile/gutenberg-mobile/blob/9df217c30ebf208a32f7b61f2d7104ba0a876306/__device-tests__/gutenberg-editor-sanity-test-4-visual.test.js#L104-L107). Selecting the text and applying the formatting toggles the active state of the formatting buttons.

1. Do you feel this approach (using sequence of `typeTextToTextBlock` + `selectTextFromElement` + `toggleFormatting`) is adequate to cover the test case and generate the screenshot? It appears to be adequate to me, but I could be misinterpreting the result.
2. There seems to be an issue with toggling the formatting of strikethrough. [This code](https://github.com/wordpress-mobile/gutenberg-mobile/blob/2f6a456c7a970c21cd4df24bb4379bda11088294/__device-tests__/gutenberg-editor-writing-flow.test.js#L52-L67) to generate the strikethrough formatting text appears to never run, and I have been unable to determine why. Does anything look obvious to you? I've included a video of the behavior when running the test, but it may also help to run the test locally if you have a moment.

https://user-images.githubusercontent.com/643285/231393331-2f99439d-6d57-414f-bc67-770305dee31b.mov

Reply:
> Do you feel this approach (using the sequence of typeTextToTextBlock + selectTextFromElement + toggleFormatting + takeScreenshot) is adequate to cover the test case of checking that "the proper format buttons are selected when the cursor is under"? 

Yes, I do believe it is adequate. It is not exactly how I personally execute the test manually, but there will always be various ways to accomplish a given task. 

> There seems to be an issue with toggling the formatting of strikethrough. [This code](https://github.com/wordpress-mobile/gutenberg-mobile/blob/2f6a456c7a970c21cd4df24bb4379bda11088294/__device-tests__/gutenberg-editor-writing-flow.test.js#L52-L67) to generate the strikethrough formatting text appears to never run, and I have been unable to determine why. Does anything look obvious to you? I've included a video of the behavior when running the test, but it may also help to run the test locally if you have a moment.

The following error outputted when running the test states the strikethrough button element is unfound. 

```
  ● Gutenberg Editor - Writing Flow › TC007 - Test format detection under the cursor › checks that the proper format buttons are selected when the cursor is under

    [waitForElementByXPath("//XCUIElementTypeButton[@name=\"Strikethrough\"]",{},2000)] Element condition wasn't satisfied!
```

My best guess for that occurring is that the strikethrough button is hidden because it is outside the visible bounds of the horizontal scroll view. We need to manually scroll the view and then execute the `toggleFormatting` helper or update the `toggleFormatting` helper to loop between querying and scrolling until it finds the target (similar to the [`findBlockButton`](https://github.com/WordPress/gutenberg/blob/8039d57822457b2a639142375beb6dac9e6bf468/packages/react-native-editor/__device-tests__/pages/editor-page.js#L428-L496) helper).